mbed-os-example-mesh-minimal build fails with KW41Z and GCC_ARM

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************
-->

### Description of defect

mbed-os-example-mesh-minimal fails to build against Mbed OS master when using command:

`mbed compile -t GCC_ARM -m KW41Z --app-config ./configs/mesh_6lowpan.json`

The build produces error:
```
Link: mbed-os-example-mesh-minimal
BUILD/KW41Z/GCC_ARM/mbed-os/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/fsl_common.o: In function `InstallIRQHandler':
/home/arto/github/mbed-os-example-mesh-minimal/./mbed-os/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/fsl_common.c:63: undefined reference to `__Vectors'
collect2: error: ld returned 1 exit status
[ERROR] BUILD/KW41Z/GCC_ARM/mbed-os/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/fsl_common.o: In function `InstallIRQHandler':
/home/arto/github/mbed-os-example-mesh-minimal/./mbed-os/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/fsl_common.c:63: undefined reference to `__Vectors'
collect2: error: ld returned 1 exit status

```

The application builds successfully with ARM toolchain.

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->


#### Target(s) affected by this defect ?
KW41Z

#### Toolchain(s) (name and version) displaying this defect ?
GCC_ARM

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->
165be79392ae7b1bee4388d2bc8ed8281978f07c

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)


#### How is this defect reproduced ? 

- git clone git@github.com:ARMmbed/mbed-os-example-mesh-minimal.git
- cd mbed-os-example-mesh-minimal
- mbed deploy
- cd mbed-os
- git checkout master
- cd ..
- mbed compile -t GCC_ARM -m KW41Z --app-config ./configs/mesh_6lowpan.json

--- a/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/fsl_common.c
+++ b/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/fsl_common.c
@@ -29,6 +29,12 @@
 */
 
 #include "fsl_common.h"
+
+/* At least CMSIS 5.7 introduced vector table and define this GCC linker symbol to
+ __Vectors - not valid for many NXP MCU. To fix this, we undefine it if already defined
+ as it comes from a linker */
+#undef __VECTOR_TABLE
+
 void InstallIRQHandler(IRQn_Type irq, uint32_t irqHandler)
 {
 /* Addresses for VECTOR_TABLE and VECTOR_RAM come from the linker file */

